mission-control/docs/topology/references/connection.md

```diff
@@ -0,0 +1,81 @@
+# Connections
+
+Connections are an easy way to authenticate against sources. It can be created via a CRD or by adding it in the settings page
+
+
+A sample connection CRD looks like:
+```yaml
+apiVersion: mission-control.flanksource.com/v1
+kind: Connection
+metadata:
+  name: payments-database
+spec:
+  type: postgres
+  url:
+    value: 'postgres://$(username):$(password)@postgres.host.com/payments'
+  username:
+    valueFrom:
+      secretKeyRef:
+        name: payments-database-credentials
+        key: POSTGRES_USER
+  password:
+    valueFrom:
+      secretKeyRef:
+        name: payments-database-credentials
+        key: POSTGRES_PASSWORD
+```
+
+Eventually, the URL that gets templated is used for establishing connections. This can be used for any datasource that authenticates via URL (PostgreSQL, MySQL, MSSQL, Redis, Opensearch, Elasticsearch etc.)
+
+A connection string can be represented in the form of `type/connection_name` or `type/namespace/connection_name` 
```

Comment:
We also support the `id` of the connection. hmm but that's probably practical for internal use only.